### PR TITLE
Fix MapBatchTest

### PR DESCRIPTION
**Pull request by gdbelvin**
Fixed the following problems:

- NewNodeIDFromRelativeBitInt
  - Was placing subtree indexes in a left justified rather than right
    justified manner.

- NodeID.Split
  - Modular arithmetic for suffix.Bits assumed equal strata levels

### CODE REVIEW

File: integration/maptest/map_test.go [hunk 1]
```diff
@@ -366,3 +365,12 @@ func TestNonExistentLeaf(t *testing.T) {
 		}
 	}
 }
+
+// h2b converts a hex string into []byte.
```
Comment (AlCutter) [inline integration/maptest/map_test.go]: There's `testonly.MustHexDecode` which does the same thing, BTW.

File: storage/types.go [hunk 2]
```diff
@@ -299,11 +304,19 @@ func (n *NodeID) Split(prefixBytes, suffixBits int) ([]byte, Suffix) {
...
 	copy(a, n.Path)
 
+	bits := n.PrefixLenBits - prefixBytes*8
+	if bits > suffixBits {
+		panic(fmt.Sprintf("storage Split: %x(n.PrefixLenBits: %v - prefixBytes: %v *8) > %v", n.Path, n.PrefixLenBits, prefixBytes, suffixBits))
+	}
+	if bits == 0 {
+		panic(fmt.Sprintf("storage Split: %x(n.PrefixLenBits: %v - prefixBytes: %v *8) == 0", n.Path, n.PrefixLenBits, prefixBytes))
+	}
+	suffixBytes := (bits + 7) / 8
```
Comment (AlCutter) [inline storage/types.go]: you can use `bytesForBits` for this :)

Comment (Martin2112) [inline storage/types.go]: He might have removed that in a previous PR?

Comment (AlCutter) [inline storage/types.go]: It lives on (at least currently): https://github.com/google/trillian/blob/d9f0b5187c3cca04fa30a9fdc63fb3afc595fda4/storage/types.go#L71

File: integration/maptest/map_test.go [hunk 3]
```diff
@@ -284,7 +283,7 @@ func RunMapBatchTest(ctx context.Context, env *integration.MapEnv, tree *trillia
 
 		if err := verifyGetMapLeavesResponse(getResp, indexes, int64(numBatches),
 			pubKey, hasher, tree.TreeId); err != nil {
-			return err
+			return fmt.Errorf("batch: %v,  %v", i, err)
```
Comment (gdbelvin) [inline integration/maptest/map_test.go]: done

File: integration/maptest/map_test.go (hunk 1 above)
Comment (gdbelvin) [inline integration/maptest/map_test.go]: Shorter names are nice?

File: storage/types.go (hunk 2 above)
Comment (gdbelvin) [inline storage/types.go]: done

File: integration/maptest/map_test.go (hunk 1 above)
Comment (AlCutter) [inline integration/maptest/map_test.go]: So is deduplication, how about this:
```golang
var h2b = testonly.MustHexDecode
```

Comment (gdbelvin) [inline integration/maptest/map_test.go]: ohhhh, I like!

Comment (gdbelvin) [inline integration/maptest/map_test.go]: done

**Pull request merged**